### ISSUE TITLE
Escopo 4 — Otimização 3: lm_head adaptativo com detecção de entropia

### ISSUE THREAD

**Issue by fabious054** [fabious054/CandleCLI#1]
## Contexto

Terceira e última otimização do Escopo 4 (Performance). Otimizações 1 (kernel matmul Q8_0) e 2 (MLP fundido) já mergeadas — baseline atual: ~7.7 tok/s validado ao vivo (benchmark isolado: ~14.6–15.8 tok/s).

Profiling original (antes do Escopo 4) mostrava lm_head + resto em ~20% do tempo de decode (~21ms/token) — projeção pro vocabulário completo (~152k tokens) a cada passo, quando o sampler com top-k=40 descarta a esmagadora maioria.

## O que implementar

`src/kernel/lm_head.rs` — lm_head adaptativo baseado em confiança do modelo (entropia da distribuição):

- Entropia < `entropy_threshold` → modelo confiante → calcula só os top-k logits (caminho rápido)
- Entropia >= `entropy_threshold` → modelo inseguro → calcula o vocabulário completo (caminho preciso, preserva top-p exato)

Novo campo em `config.toml`:
```toml
[behavior]
max_new_tokens = 512
entropy_threshold = 2.0
```

Integração em `arch/qwen/layers.rs`, substituindo a chamada de lm_head atual.

## Critérios de aceite

- [ ] `src/kernel/lm_head.rs` implementado do zero (sem `candle_nn::ops`/`candle_transformers`)
- [ ] `entropy_threshold` configurável via `config.toml`, default 2.0
- [ ] Integrado em `arch/qwen/layers.rs`
- [ ] `cargo test --lib` — 25/25 (mais os testes novos do lm_head)
- [ ] Benchmark isolado (prefill 32 tokens + 30 decode) medido antes/depois
- [ ] Validação manual ao vivo pelo Fabio (tok/s real + qualidade de resposta)

## Referências

- ADR-0003 (threshold peça 7), ADR-0004 (precisão matmul) — mesmo espírito de documentar divergências numéricas conscientes
- Card Trello: diretriz de performance "máximo do hardware, mínimo de esforço"

**Comment by fabious054** [comment]
## Parte A implementada — pronta para os testes locais

Branch `escopo4/otimizacao-3-lmhead` sincronizada com `main` (estava desatualizada, criada antes do push das Otimizações 1+2) e com a mudança da Parte A aplicada por cima, em 8 commits:

1. `a7e5377` a `f01f74c` — sync com `main`: Cargo.toml, lib.rs, kernel/mod.rs, kernel/matmul.rs, kernel/mlp.rs, attention.rs, arch/qwen/mlp.rs (conteúdo idêntico ao que já está aprovado em main)
2. `e00c739` — feat: Otimização 3 Parte A — lm_head via Q8_0Matmul kernel (o commit que importa)

### O que mudou de fato (só src/arch/qwen/layers.rs)

Model::lm_head era QMatMul (candle), agora é Q8_0Matmul (nosso, da Otimização 1). Estruturalmente igual a qualquer outra projeção — não precisou de código novo no kernel, só trocar o tipo. forward() não mudou.

### Risco real a observar

lm_head agora tem a mesma divergência de precisão que o ADR-0004 documenta pra atenção/MLP (nossa ativação em F32 vs oráculo quantizando a ativação também). Terceira fonte de divergência empilhada sobre o ADR-0003. O teste model_matches_oracle_end_to_end (threshold 0.24) já tinha margem apertada no prefill (~0.018). Pode estourar agora — se estourar, não é bug novo, é o mesmo fenômeno do ADR-0004, conserto é subir o threshold.

### Para rodar

```
git fetch origin
git checkout escopo4/otimizacao-3-lmhead
git pull origin escopo4/otimizacao-3-lmhead
cargo test --lib
```

Se falhar por causa do threshold 0.24: me avisa o valor exato do diff que eu ajusto com o ADR atualizado.

Depois, com testes passando: cargo run --release, observar o tok/s.
